### ISSUE TITLE
Document the lack of DTC on .NET Core

### ISSUE THREAD

**Issue by ArveSystad** [dotnet/EntityFramework.Docs#1122]
It would be nice with an overview of how EF Core deals with transactions _compared to EF6_ somewhere. In the middle of porting a lot of code now, and we're getting different behavior when having multiple SQL Connections (with the same connection string) within one omnipresent transaction (in an integration test). After moving to EFCore (but still on .Net framework., for now), we seem to end up in DTC the moment we do `new SqlConnection` in addition to the EFCore one. Before we started the port, we had several connections going, no problem. I can't find any information about how this differs from EF6, so it's a bit of a black box.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1073c238-9389-d826-a696-f1108a4f2ce0
* Version Independent ID: cf56babd-def0-d76e-8eee-e33341b6a009
* Content: [Transactions - EF Core](https://docs.microsoft.com/en-us/ef/core/saving/transactions)
* Content Source: [entity-framework/core/saving/transactions.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/saving/transactions.md)
* Product: **entity-framework**
* GitHub Login: @rowanmiller
* Microsoft Alias: **divega**

**Comment by divega** [comment]
@ArveSystad if something like this is happening, it sounds like worth reporting as a product issue at https://github.com/aspnet/EntityFrameworkCore/issues/new rather than here in the documentation. The main thing we did differently in EF Core is to avoid closing the connection if it already enlisted to a System.Transactions transaction, which should actually avoid escalating to DTC in some common scenarios, but maybe we missed something else.

**Comment by ArveSystad** [comment]
Ok. Will take a closer look, and see if I can reproduce with a minimal example at some point. I might have missed out on other changes that have been made that might be the origin. This is useful information regardless. Thanks!

**Comment by Vclcl** [comment]
This code escalating to DTC in EF Core but avoid escalating in EF 6. What do you think why?
```
using (var transaction = new TransactionScope())
{
	using (var context = new BaseDbContext())
	{
		context.Entities.Add(new Entity { Name = "test entity" });

		context.SaveChanges();

		using (var context2 = new BaseDbContext())
		{
			context2.Entities.Add(new Entity { Name = "test entity" });

			context2.SaveChanges();
		}
	}

	transaction.Complete();
}
```

I assume because the connection does not return to the pool after the SaveChanges.

**Comment by divega** [comment]
Yes, I think it is the same reason @ArveSystad found before. The workaround in this case would be to pass the connection from the outer to the nested DbContext. 

BTW, we have fixed this in EF Core 3.0 by going back to the EF6 behavior.
